### ISSUE TITLE
Pipeline: message Jenkins doesn’t have label xxx is not clear enough

### ISSUE THREAD

**Issue by ppitonak** [openshiftio/openshift.io#3797]
1. Start a new build
2. Open Jenkins log

result:
* The job will stop for some time after the following message is printed
```
Still waiting to schedule task
Jenkins doesn’t have label mavenprojectname_2
```
It's not clear if it's an expected behavior or some error and whether it's fatal or will continue after some time.

**Comment by stevengutz** [comment]
This is definitely not a SEV2

**Comment by piyush1594** [comment]
@rupalibehera @jaseemabid Can you please check if we can do anything related to this?

**Comment by piyush1594** [comment]
@ppitonak Its something like its waiting for a slave with that label and as soon as slave comes up it starts executing the job. Correct me if I am wrong @rupalibehera @jaseemabid

**Comment by jaseemabid** [comment]
@piyush1594 Yep, it's the k8s plugin waiting for agents to be available. We could add a better message there, like "please wait" because that's all you can do as of now.

**Comment by piyush1594** [comment]
@jaseemabid Then I think this is not an issue. @ppitonak WDYT?

**Comment by ppitonak** [comment]
It is a usability issue... when a user sees this message, they don't know if it's expected behavior or an error and when it is expected to continue.

**Comment by rupalibehera** [comment]
This is not a sev-2, and this is the kubernetes plugin as mentioned by the @jaseemabid , I am not sure how much we can do to prevent this on the Jenkins console output, may be on the OSIO UI we can say that "slave is not ready yet" until the slave is assigned.

**Comment by sthaha** [comment]
downgrading the severity to 3 after referring https://github.com/openshiftio/openshift.io/wiki/Labels 

> Severity 2 (high) A problem where the software is functioning but your use in a production environment is severely reduced. The situation is causing a high impact to portions of your business operations and no procedural workaround exists. 

IMHO the bug doesn't meet that criteria.

**Comment by sthaha** [comment]
I am setting p2 so that we can work on it next sprint

**Comment by ppitonak** [comment]
@sthaha there are two related issues

#3797 is UX issue... message is confusing/not clear enough
#4020 pipeline fails, the last thing in Jenkins log is this message

IMHO #4020 should be P2, #3797 is nice to have

**Comment by sthaha** [comment]
@ppitonak both added to our backlog for next sprint https://github.com/orgs/fabric8io/projects/2?fullscreen=true

**Comment by hrishin** [comment]
This message is coming from Jenkins 
https://github.com/jenkinsci/jenkins/blob/22aa2e6e766074d11249893e3f35e0b99e20d3d0/core/src/main/resources/hudson/model/Messages.properties#L346

To fix such kind of issues. Either we fix this on Jenkins side or need to check plugin or Jenkins provides a timeout message.

**Comment by piyush-garg** [comment]
@ppitonak 

Now the Jenkins log looks like this 

```
Loaded the useDockerSocket flag false
[Pipeline] podTemplate
[Pipeline] {
[Pipeline] node
Still waiting to schedule task
All nodes of label ‘maventest_2_9’ are offline
Agent jenkins-slave-lw51d-c0th5 is provisioned from template Kubernetes Pod Template
Agent specification [Kubernetes Pod Template] (maventest_2_9): 
* [jnlp] openshift/jenkins-slave-base-centos7:v4.0(resourceLimitMemory: 256Mi)
* [maven] openshift/jenkins-slave-maven-centos7:v4.0(resourceLimitMemory: 640Mi)

Running on jenkins-slave-lw51d-c0th5 in /home/jenkins/workspace/yush-garg_test-2_master-OXFFZYTJMZT6LBEZQQK7JWJ764GSOBD5LSLYFXXQJ43P32WK
```

**Comment by ppitonak** [comment]
@piyush-garg that is much better. Is it already in prod?

**Comment by piyush-garg** [comment]
@ppitonak Yes, I have seen that in prod.
